viewing_party/party.py

```diff
@@ -1,23 +1,183 @@
+    if movie_title == None or genre == None or rating == None:
+        return None
+    movie = {"title":movie_title, "genre":genre, "rating":rating}
+    return movie
 
+def add_to_watched(user_data, movie):
+    if movie["title"] not in user_data["watched"]:
+        user_data["watched"].append(movie)
+    return user_data
+    
+def add_to_watchlist(user_data, movie):
+    if movie["title"] not in user_data["watchlist"]:
+        user_data["watchlist"].append(movie)
+    return user_data
+    
+def watch_movie(user_data, movie_title):
+    for movie in user_data["watchlist"]:
+        if movie_title == movie["title"] and movie_title not in user_data["watched"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating = 0.0
+    count = 0
+
+    if not user_data:
+        return None
+
+    for movies in user_data.values():
+        if not movies:
+            average_rating = 0.0
+            return average_rating
+        
+        for movie in movies:
+            count += 1
+            rating += movie["rating"]
+    
+    rating = rating/count
+    
+    return rating
+
+def get_most_watched_genre(user_data):
+    genre_count = {}
+    most_watched_genre = ""
+    most_watched = 0
+
+    if not user_data:
+        return None
+    
+    for movies in user_data.values():
+        if not movies:
+            return None
+        
+        for movie in movies:
+            if 'genre' in movie:
+                genre = movie["genre"]
+                if genre not in genre_count:
+                    genre_count[genre] = 1
+                else:
+                    genre_count[genre] += 1
+                most_watched = genre_count[genre]
+    
+    for genre in genre_count:
+        if genre_count[genre] > most_watched:
+            most_watched = genre_count[genre]
+            most_watched_genre = genre
+
+    return most_watched_genre
+
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    user_watched = []
+    friend_watched = []  
+    user_unique_watched = []
+
+    for watched_movie in user_data['friends']:
+        for movie in watched_movie["watched"]:
+            friend_watched.append(movie)
+    for movie in user_data["watched"]:
+        user_watched.append(movie)
+    for movie in user_watched:
+        if movie not in friend_watched:
+            user_unique_watched.append(movie)
+
+    return user_unique_watched    
+
+def get_friends_unique_watched(user_data):
+    user_watched = []
+    friend_watched = []  
+    friends_unique_watched = []
+
+    for watched_movies in user_data['friends']:
+        for movie in watched_movies["watched"]:

```

Comment:
`user_data['friends']` returns a list of dictionaries and each of the dictionaries represent a friend. Within each friend dictionary are more movie dictionaries. 

The looping variable on line 102 `watched_movies` doesn't quite accurately describe that the element we're looping over is a friend. 

Consider renaming the variables so they're more descriptive which would make your code more self-documenting!
```suggestion
    for friend in user_data['friends']:
        for friend_movie in friend["watched"]:
```